pkg/api/mlflow/service/artifact/validators.go

```diff
@@ -9,7 +9,7 @@ import (
 )
 
 // RelativePathRegExp checks for the `../` in provided path.
-var RelativePathRegExp = regexp.MustCompile(`\.{2}\/|^\.{2}$`)
+var RelativePathRegExp = regexp.MustCompile(`\/\.{2}|\.{2}\/|^\.{2}`)
```

Comment:
I think this is still not ok, as paths like `foo/..bar` or `foo../bar` would be considered invalid. I may be wrong, but I think that the regex is becoming really complex and it may be simpler to just check for any directories named exactly `..` in the path.

Reply:
no, this paths are fully valid. add them to unit tests.

Reply:
I'm very sorry Dmitriy, but the unit tests are wrong. You are setting the `RunID` instead of the `Path`!

Reply:
yes, Im a bit blinded already. now all good and tests still ok.

Reply:
sorry but the tests are still wrong — you are not passing the request to `ValidateListArtifactsRequest`!

Reply:
ohh, you are right. I have to disable my rush when you are online and answer so fast.

Reply:
then, yep. fixed and extend tests. looks good now.

Reply:
yep, you are right.  I have to stop rushing when you online and answer so fast. no should be all good. yes, probably `regexp` is not right(simplest) way to use it.